### PR TITLE
Added buses on map

### PR DESCRIPTION
**Pull request by hiba9201**
- Replaced ettu api request to our backend
- Added some animations
- Added model, depo, first and last station and state number to vehicle sidebar

### CODE REVIEW

File: common/types/masstrans.ts [hunk 1]
```diff
@@ -35,3 +35,20 @@ export type Stop = {
...
+export type Unit = {
+    id: number;
+    num: string;
+    depoTitle: string;
+    firstStation: string;
+    lastStation: string;
+    routeDirection: string;
+    type: ClientUnit;
+    course: number;
+    accessibility: number;
```
Comment (a-chabin) [inline common/types/masstrans.ts:48]: Probably `boolean` here?

File: client/components/Map/Stations/Item/MapStationsItem.module.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.MapStationsItemIcon {
+    transition: height 150ms ease-out, width 150ms ease-out, margin 150ms ease-out;
+    will-change: height, width, margin;
```
Comment (a-chabin) [inline client/components/Map/Stations/Item/MapStationsItem.module.css:3]: Are there many stops here? Doesn't that reduce performance?

https://w3c.github.io/csswg-drafts/css-will-change/#using

File: client/components/Map/Vehicles/Sidebar/MapVehiclesSidebar.tsx [hunk 3]
```diff
@@ -62,22 +64,55 @@ export function MapVehiclesSidebar({
...
+        if (splittedStateNum.length === 1) {
+            return null;
+        }
+
+        const roadNumber = splittedStateNum.slice(0, -1);
+        let region = splittedStateNum.at(-1);
+
+        if (!parseInt(region, 10)) {
+            roadNumber.push(region);
+            region = '196';
```
Comment (a-chabin) [inline client/components/Map/Vehicles/Sidebar/MapVehiclesSidebar.tsx:87]: Let's move to `const SVERDLOVSK_REGION = 196`

File: common/types/masstrans.ts (hunk 1 above)
Comment (a-chabin) [reply common/types/masstrans.ts:48]: Or is it a contract from `masstrans'?

File: client/components/Map/Stations/Item/MapStationsItem.module.css (hunk 2 above)
Comment (a-chabin) [reply client/components/Map/Stations/Item/MapStationsItem.module.css:3]: Reduce the use of rules that triggers repaints

`width, height, margin, top/right/bottom/left` → `transform `

Comment (a-chabin) [reply client/components/Map/Stations/Item/MapStationsItem.module.css:3]: Move it to issue
https://github.com/ekaterinburgdev/transport/issues/33

Comment (hiba9201) [reply client/components/Map/Stations/Item/MapStationsItem.module.css:3]: Unfortunately for stops we use default icons api from leaflet and they change height and width when we change size(
I think we should really look into it in separate issue

**Review by a-chabin — APPROVED** [review]

**Review by a-chabin — APPROVED** [review]

**Pull request merged**